mdata/aggmetric.go

```diff
@@ -66,6 +66,12 @@ func NewAggMetric(store Store, cachePusher cache.CachePusher, key string, retent
 	return &m
 }
 
+func (a *AggMetric) maybeSyncChunkSaveState(ts uint32) {
```

Comment:
"maybe*" is so cryptic. how is someone who reads this code supposed to know what maybeSyncChunkSaveState means.  Under what conditions might this function save the chunkSaveState and under what conditions might it not.